src/sumo_store_riak.erl

```diff
@@ -417,39 +424,57 @@ validate_date({FieldType, _, FieldValue}) ->
 
 %% @private
 sleep(Doc) ->
-  sumo_utils:doc_transform(fun sleep_fun/1, Doc).
-
+  sumo_utils:doc_transform(fun sleep_fun/4, Doc).
 %% @private
-sleep_fun({_, FieldName, undefined}) when FieldName /= id ->
+sleep_fun(_, FieldName, undefined, _) when FieldName /= id ->
   <<"$nil">>;
-sleep_fun({FieldType, _, FieldValue})
+sleep_fun(FieldType, _, FieldValue, _)
     when FieldType =:= datetime; FieldType =:= date ->
   case {FieldType, sumo_utils:is_datetime(FieldValue)} of
     {datetime, true} -> iso8601:format(FieldValue);
     {date, true}     -> iso8601:format({FieldValue, {0, 0, 0}});
     _                -> FieldValue
   end;
-sleep_fun({_, _, FieldValue}) ->
+sleep_fun(custom, _, FieldValue, FieldAttrs) ->
+  Type = sumo_utils:keyfind(type, FieldAttrs, custom),
+  sleep_custom(FieldValue, Type);
+sleep_fun(_, _, FieldValue, _) ->
   FieldValue.
 
 %% @private
-wakeup(Doc) ->
-  sumo_utils:doc_transform(fun wakeup_fun/1, Doc).
+sleep_custom(FieldValue, term) ->
+  base64:encode(term_to_binary(FieldValue));
+sleep_custom(FieldValue, _) ->
```

Comment:
Doesn't this defeat the nested field search capabilities of this backend? Shouldn't we implement something a little bit smarter to store nested docs?